File: CalculadoraJava/app/src/main/java/com/jairomejia/calculadorajava/MainActivity.java
package com.jairomejia.calculadorajava;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    EditText editTextValorA, editTextValorB;
    EditText editTextResultadoC;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        editTextValorA = findViewById(R.id.editTextTituloValorA);
        editTextValorB = findViewById(R.id.editTextTituloValorB);
        editTextResultadoC = findViewById(R.id.editTextTituloResultado);

        Button btnSuma = findViewById(R.id.buttonSumar);
        Button btnResta = findViewById(R.id.buttonRestar);
        Button btnMultiplicacion = findViewById(R.id.buttonMultiplicar);
        Button btnDivision = findViewById(R.id.buttonDividir);

        btnSuma.setOnClickListener(v -> suma());
        btnResta.setOnClickListener(v -> resta());
        btnMultiplicacion.setOnClickListener(v -> multiplicacion());
        btnDivision.setOnClickListener(v -> division());

    }
    private void suma()
    {
        try {
            String stringValorA = editTextValorA.getText().toString();
            String stringValorB = editTextValorB.getText().toString();

            if (stringValorA.isEmpty() || stringValorB.isEmpty())
            {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show();
            }
            else {
                double doubleValorA = Double.parseDouble(stringValorA);
                double doubleValorb = Double.parseDouble(stringValorB);
                double doubleResultadoC = 0;

                doubleResultadoC = doubleValorA + doubleValorb;

                editTextResultadoC.setText(" = " + doubleResultadoC);
            }

        } catch (NumberFormatException exception){
            Toast.makeText(this, "Error: "+ exception, Toast.LENGTH_SHORT).show();
        }
    }

    private void resta()
    {
        try {
            String stringValorA = editTextValorA.getText().toString();
            String stringValorB = editTextValorB.getText().toString();

            if (stringValorA.isEmpty() || stringValorB.isEmpty())
            {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show();
            }
            else {
                double doubleValorA = Double.parseDouble(stringValorA);
                double doubleValorb = Double.parseDouble(stringValorB);
                double doubleResultadoC = 0;

                doubleResultadoC = doubleValorA - doubleValorb;

                editTextResultadoC.setText(" = " + doubleResultadoC);
            }

        } catch (NumberFormatException exception){
            Toast.makeText(this, "Error: "+ exception, Toast.LENGTH_SHORT).show();
        }
    }
    
    private void multiplicacion()
    {
        try {
            String stringValorA = editTextValorA.getText().toString();
            String stringValorB = editTextValorB.getText().toString();

            if (stringValorA.isEmpty() || stringValorB.isEmpty())
            {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show();
            }
            else {
                double doubleValorA = Double.parseDouble(stringValorA);
                double doubleValorb = Double.parseDouble(stringValorB);
                double doubleResultadoC = 0;

                doubleResultadoC = doubleValorA * doubleValorb;

                editTextResultadoC.setText(" = " + doubleResultadoC);
            }

        } catch (NumberFormatException exception){
            Toast.makeText(this, "Error: "+ exception, Toast.LENGTH_SHORT).show();
        }
    }
    private void division()
    {
        try {
            String stringValorA = editTextValorA.getText().toString();
            String stringValorB = editTextValorB.getText().toString();

            if (stringValorA.isEmpty() || stringValorB.isEmpty())
            {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show();
            } else if (stringValorB.equals("0")) {
                Toast.makeText(this, "No se puede dividir por 0", Toast.LENGTH_SHORT).show();
            } else {
                double doubleValorA = Double.parseDouble(stringValorA);
                double doubleValorb = Double.parseDouble(stringValorB);
                double doubleResultadoC = 0;

                doubleResultadoC = doubleValorA / doubleValorb;

                editTextResultadoC.setText(" = " + doubleResultadoC);
            }

        } catch (NumberFormatException exception){
            Toast.makeText(this, "Error: "+ exception, Toast.LENGTH_SHORT).show();
        }
    }
}

File: CalculadoraKotlin/app/src/main/java/com/jairomejia/calculadorakotlin/MainActivity.kt
package com.jairomejia.calculadorakotlin

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    var editTextValorA: EditText? = null
    var editTextValorB: EditText? = null
    var editTextResultadoC: EditText? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        editTextValorA = findViewById<EditText?>(R.id.editTextTituloValorA)
        editTextValorB = findViewById<EditText?>(R.id.editTextTituloValorB)
        editTextResultadoC = findViewById<EditText?>(R.id.editTextTituloResultado)

        val btnSuma = findViewById<Button?>(R.id.buttonSumar)
        val btnResta = findViewById<Button?>(R.id.buttonRestar)
        val btnMultiplicacion = findViewById<Button?>(R.id.buttonMultiplicar)
        val btnDivision = findViewById<Button?>(R.id.buttonDividir)

        btnSuma.setOnClickListener(View.OnClickListener { v: View? -> suma() })
        btnResta.setOnClickListener(View.OnClickListener { v: View? -> resta() })
        btnMultiplicacion.setOnClickListener(View.OnClickListener { v: View? -> multiplicacion() })
        btnDivision.setOnClickListener(View.OnClickListener { v: View? -> division() })
    }
    private fun suma() {
        try {
            val stringValorA = editTextValorA!!.getText().toString()
            val stringValorB = editTextValorB!!.getText().toString()

            if (stringValorA.isEmpty() || stringValorB.isEmpty()) {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show()
            } else {
                val doubleValorA = stringValorA.toDouble()
                val doubleValorb = stringValorB.toDouble()
                var doubleResultadoC = 0.0

                doubleResultadoC = doubleValorA + doubleValorb

                editTextResultadoC!!.setText(" = " + doubleResultadoC)
            }
        } catch (exception: NumberFormatException) {
            Toast.makeText(this, "Error: " + exception, Toast.LENGTH_SHORT).show()
        }
    }

    private fun resta() {
        try {
            val stringValorA = editTextValorA!!.getText().toString()
            val stringValorB = editTextValorB!!.getText().toString()

            if (stringValorA.isEmpty() || stringValorB.isEmpty()) {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show()
            } else {
                val doubleValorA = stringValorA.toDouble()
                val doubleValorb = stringValorB.toDouble()
                var doubleResultadoC = 0.0

                doubleResultadoC = doubleValorA - doubleValorb

                editTextResultadoC!!.setText(" = " + doubleResultadoC)
            }
        } catch (exception: NumberFormatException) {
            Toast.makeText(this, "Error: " + exception, Toast.LENGTH_SHORT).show()
        }
    }

    private fun multiplicacion() {
        try {
            val stringValorA = editTextValorA!!.getText().toString()
            val stringValorB = editTextValorB!!.getText().toString()

            if (stringValorA.isEmpty() || stringValorB.isEmpty()) {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show()
            } else {
                val doubleValorA = stringValorA.toDouble()
                val doubleValorb = stringValorB.toDouble()
                var doubleResultadoC = 0.0

                doubleResultadoC = doubleValorA * doubleValorb

                editTextResultadoC!!.setText(" = " + doubleResultadoC)
            }
        } catch (exception: NumberFormatException) {
            Toast.makeText(this, "Error: " + exception, Toast.LENGTH_SHORT).show()
        }
    }

    private fun division() {
        try {
            val stringValorA = editTextValorA!!.getText().toString()
            val stringValorB = editTextValorB!!.getText().toString()

            if (stringValorA.isEmpty() || stringValorB.isEmpty()) {
                Toast.makeText(this, "Campos vacíos", Toast.LENGTH_SHORT).show()
            } else if (stringValorB == "0") {
                Toast.makeText(this, "No se puede dividir por 0", Toast.LENGTH_SHORT).show()
            } else {
                val doubleValorA = stringValorA.toDouble()
                val doubleValorb = stringValorB.toDouble()
                var doubleResultadoC = 0.0

                doubleResultadoC = doubleValorA / doubleValorb

                editTextResultadoC!!.setText(" = " + doubleResultadoC)
            }
        } catch (exception: NumberFormatException) {
            Toast.makeText(this, "Error: " + exception, Toast.LENGTH_SHORT).show()
        }
    }

}